clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -178,4 +192,55 @@ public static Properties producerConfig(final String bootstrapServers,
         properties.putAll(additional);
         return properties;
     }
+
+    public static Properties producerConfig(final String bootstrapServers, Class keySerializer, Class valueSerializer) {
+        return producerConfig(bootstrapServers, keySerializer, valueSerializer, new Properties());
+    }
+
+    public static Properties consumerConfig(final String bootstrapServers,
+                                            final String groupId,
+                                            final Class keyDeserializer,
+                                            final Class valueDeserializer,
+                                            final Properties additional) {
+
+        final Properties consumerConfig = new Properties();
+        consumerConfig.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        consumerConfig.put(ConsumerConfig.GROUP_ID_CONFIG, groupId);
+        consumerConfig.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        consumerConfig.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, keyDeserializer);
+        consumerConfig.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, valueDeserializer);
+        consumerConfig.putAll(additional);
+        return consumerConfig;
+    }
+
+    /**
+     * returns consumer config with random UUID for the Group ID
+     */
+    public static Properties consumerConfig(final String bootstrapServers, Class keyDeserializer, Class valueDeserializer) {
+        return consumerConfig(bootstrapServers,
+                              UUID.randomUUID().toString(),
+                              keyDeserializer,
+                              valueDeserializer,
+                              new Properties());
+    }
+
+    /**
+     *  uses default value of 30 seconds for timeout
+     */
+    public static void waitForCondition(TestCondition testCondition) throws InterruptedException {
+        waitForCondition(testCondition, 30000);
+    }
+
+    /**
+     *  Used to wait for specific conditions/state to be me during a test
+     *  this is meant to be a replacement for using Thread.sleep
+     */
+    public static void waitForCondition(TestCondition testCondition, long maxTimeMillis) throws InterruptedException {
+        long startTime = System.currentTimeMillis();
+
+        while (!testCondition.conditionMet() && ((System.currentTimeMillis() - startTime) < maxTimeMillis)) {
+            Thread.sleep(Math.min(maxTimeMillis, 100L));
+        }
```

Comment:
> What is the reasoning for not throwing an exception if the condition is not met after the timeout?

@ijuma  @guozhangwang 

My reasoning is two fold:

- Halting the progress of a test may not always yield an "assertable" condition

- Secondly I feel its more clear (at least in some cases) to have the "assertion" phase after `waitForCondition` returns (lines 200 - 208 of `RegexSourceIntegrationTest.testRegexMatchesTopicsAWhenDeleted` for example)

But I agree with you in that doing so would make the tests more concise.  What do you think about changing `waitForCondition` to take a third parameter `String errorMessage` and do the following when breaking out of the loop
```
if ( !testCondtion.conditionMet() && errorMessage != null ) {
     throw new AssertionError(errorMessage);
}
```